### DISCUSSION TITLE
Storage batch upload, download and upback

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#1930] [Questions] [answered]
I am considering using supabase storage to replace my S3, but I want to know first:
1. Is supabase storage backed by s3?
2. If yes, can I get access to the s3 bucket directly without using supabase?
3. Can I batch upload a large number of files (e.g. 100000 files) to supabase storage?
4. Can I batch download a large number of files (e.g. 100000 files) to supabase storage?
5. I am feeling unsafe to put my valuable files on supabase storage. How can I backup all the files from Storage to another place, e.g. local or a personal s3 bucket?
Can anyone help me out on my decision?

**Comment by inian** [comment, marked as answer]
Hey @d-mok,

> Is supabase storage backed by s3?

Yes, so you get all the durability guarantees of S3

> If yes, can I get access to the s3 bucket directly without using supabase?

Not at the moment. The underlying S3 bucket is multi-tenant and pretty locked down from external access. 

> Can I batch upload a large number of files (e.g. 100000 files) to supabase storage?

Yes, you can. As long as your [pricing plan](https://supabase.io/pricing) allows it. 

> Can I batch download a large number of files (e.g. 100000 files) to supabase storage?

Yes, you can. As long as your [pricing plan](https://supabase.io/pricing) allows it. 

> I am feeling unsafe to put my valuable files on supabase storage. How can I backup all the files from Storage to another place, e.g. local or a personal s3 bucket?

Hmm, this is an interesting idea. S3 has a replication feature which we can use to replicate the underlying objects to your own bucket. Creating an issue [here](https://github.com/supabase/storage-api/issues/27)


Ping beta@supabase.io if there are any issues specific to your project.
